### DISCUSSION TITLE
Menu Scroll with Limited Amount of Lines

### DISCUSSION THREAD

**Discussion by nivzelber** [marlonrichert/zsh-autocomplete#699] [Q&A] [answered]
I'd like a combination of settings in ~/.zshrc like
```
zstyle ':autocomplete:history-search-backward:*' list-lines 10
# another setting here
```

Where the second setting makes pressing the up arrow on the first suggestion (10th from the bottom) will continue to the previous suggestion (i.e. 11th one) instead of circling back to start.  
Is there an option for that?

**Comment by marlonrichert** [comment]
[As mentioned in the Readme](https://github.com/marlonrichert/zsh-autocomplete/blob/main/README.md#change-the-max-number-of-lines-shown), this will let you scroll up to 2000 lines backward:
```sh
zstyle ':autocomplete:history-search-backward:*' list-lines 2000
```

You could even decide to make it scroll through all of history:
```sh
zstyle -e ':autocomplete:history-search-backward:*' list-lines '$HISTSIZE'
```
However, depending on the actual number of items in your history, this can be _extremely slow._ That is, unfortunately, a limitation of Zsh and there's not much I can do about it.

**Reply by nivzelber** [reply]
This is not what I'm looking for as setting a really high number will make the menu occupy the entire screen.  
I want a limited list, with scrolling ability instead of cycling back to the first suggestion from the last one. If I set the number to, let's say, 5, I want to always see 5 suggestions at most on the screen, but with the ability to scroll to other suggestions (i.e. 1-5, 6-10, 11-15, ...).

**Reply by max0l** [reply]
yes, I was also about to ask about this

**Reply by marlonrichert** [reply, marked as answer]
Sorry, not natively possible in Zsh

**Reply by max0l** [reply]
I'm trying to understand this better, but I can't get my head around why it works if there are more lines that can be displayed than the actual lines on the terminal screen. If this happens, I can "scroll" through the entire history.

**Reply by eaglgenes101** [reply]
What kind of new feature would make it possible to implement such a thing in zsh?

**Reply by Hadeelghanm** [reply]
lhope My balance is big and tall
